feature/experimental/p4rt/ate_tests/base_p4rt/base_p4rt_test.go

```diff
@@ -0,0 +1,413 @@
+		})
+		if err != nil {
+			return errors.New("Errors seen when sending SetForwardingPipelineConfig.")
+		}
+		// Compare P4Info from GetForwardingPipelineConfig and SetForwardingPipelineConfig
+		getp4Info, _ := json.Marshal(resp.Config.P4Info)
+		if cmp.Equal(string(setp4Info), string(getp4Info)) {
+			fmt.Println("P4info matches fine for client", index)
+		} else {
+			err := fmt.Errorf("P4info does not match for client %d", index)
+			fmt.Println(err.Error())
+		}
+	}
+	return nil
+}
+
+// Function to compare and check if the expected table is present in RPC ReadResponse
+func verifyReadReceiveMatch(expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
+
+	matches := 0
+	for _, table := range received_entry.Entities {
+		if cmp.Equal(table.Entity, expected_update[0].Entity.Entity) {
+			fmt.Println("Table match succesful")
+			matches += 1
+			break
+		}
+	}
+	if matches == 0 {
+		return errors.New("match unsuccesful")
+	}
+	return nil
+}
+
+// TestP4rtConnect connects to the P4Runtime server over grpc
+// It then calls setupP4RTClient which sets the arbitration request and sends SetForwardingPipelineConfig with P4Info
+func TestP4rtConnect(t *testing.T) {
+
+	dut := ondatra.DUT(t, "dut")
+	ctx := context.Background()
+	ate := ondatra.ATE(t, "ate")
+
+	// configure DUT with P4RT node-id and ids on different FAPs
+	configureDUT(t, dut, []attrs.Attributes{dutPort1, dutPort2}, []string{*comp1name, *comp2name}, []uint64{deviceId1, deviceId2}, []uint32{portId1, portId2})
+
+	top := configureATE(t, ate)
+
+	// Setup two different clients for different FAPs
+	client1 := p4rt_client.P4RTClient{}
+	if err := client1.P4rtClientSet(dut.RawAPIs().P4RT(t)); err != nil {
+		t.Fatalf("Could not initialize p4rt client: %v", err)
+	}
+
+	client2 := p4rt_client.P4RTClient{}
+	if err := client2.P4rtClientSet(dut.RawAPIs().P4RT(t)); err != nil {
+		t.Fatalf("Could not initialize p4rt client: %v", err)
+	}
+
+	args := &testArgs{
+		ctx:     ctx,
+		client1: &client1,
+		client2: &client2,
+		dut:     dut,
+		ate:     ate,
+		top:     top,
+	}
+
+	if err := setupP4RTClient(ctx, args); err != nil {
+		t.Fatalf("Could not setup p4rt client: %v", err)
+	}
+
+	// RPC Write to write the table entries to the P4RT Server
+	deviceid_list := []uint64{deviceId1, deviceId2}
+	clients := []*p4rt_client.P4RTClient{args.client1, args.client2}
+	for index, client := range clients {
+		err := client.Write(&p4_v1.WriteRequest{
+			DeviceId:   deviceid_list[index],
+			ElectionId: &p4_v1.Uint128{High: uint64(0), Low: electionId},
+			Updates: wbb.ACLWbbIngressTableEntryGet([]*wbb.ACLWbbIngressTableEntryInfo{
+				{
+					Type:          p4_v1.Update_INSERT,
+					EtherType:     0x6007,
+					EtherTypeMask: 0xFFFF,
+				},
+				{
+					Type:          p4_v1.Update_INSERT,
+					EtherType:     0x88cc,
+					EtherTypeMask: 0xFFFF,
+				},
+				{
+					Type:    p4_v1.Update_INSERT,
+					IsIpv4:  0x1,
+					TTL:     0x1,
+					TTLMask: 0xFF,
+				},
+			}),
```

Comment:
Can we Add Priority: to these updates , can use same priority . example 
https://github.com/openconfig/featureprofiles/blob/35b3a5a080c03726fe0b4f4fcd5722b2aeacdef6/feature/experimental/p4rt/ate_tests/google_discovery_protocol_packetin_test/google_discovery_protocol_packetin_test.go#L466

This is as per  spec https://p4.org/p4-spec/p4runtime/main/P4Runtime-Spec.html
_"The priority field must be set to a non-zero value if the match key includes a ternary match (i.e.in the case of PSA if the P4Info entry for the table indicates that one or more of its match fields has an OPTIONAL, TERNARY or RANGE match type) or to zero otherwise. A higher priority number indicates that the entry must be given higher priority when performing a table lookup. Clients must allow multiple entries to be added with the same priority value. If a packet can match multiple entries with the same priority, it is not deterministic in the data plane which entry a packet will match. If a client wishes to make the matching behavior deterministic, it must use different priority values for any pair of table entries that the same packet matches."_

Reply:
@vbhan-cisco , thanks for the comment. 
Can we keep the priority same for all the table entries that we send in the update? I dont think there priority among  GDP, LLDP and traceroute table entries, so we can set all to 1.

Reply:
It can be the same priority.

Reply:
Done